command/server/config.go

```diff
@@ -14,12 +14,15 @@ import (
 
 // Config is the configuration for the vault server.
 type Config struct {
-	Listeners []*Listener `hcl:"-"`
-	Backend   *Backend    `hcl:"-"`
+	Listeners []*Listener    `hcl:"-"`
+	Backend   *Backend       `hcl:"-"`
 
-	DisableMlock bool   `hcl:"disable_mlock"`
+	DisableMlock bool        `hcl:"disable_mlock"`
 
-	Telemetry    *Telemetry `hcl:"telemetry"`
+	Telemetry *Telemetry     `hcl:"telemetry"`
+
+	MaxLeaseDuration int     `hcl:"max_lease_duration"`
```

Comment:
Intstead of an `int` we should use `time.Duration`. A common trick we use is to have:
  
    MaxLeaseDuration time.Duration `hcl:"-"`
    MaxLeaseDurationRaw string `hcl:"max_lease_duration"`

Then you use time.ParseDuration if the MaxLeaseDurationRaw value is there. This way you have a uniform representation and can more naturally express things as "24h" etc.